flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java

```diff
@@ -167,15 +166,11 @@ public WindowedStream(KeyedStream<T, K> input,
 					evictor).enableSetProcessingTime(setProcessingTime);
 
 		} else {
-			// we need to copy because we need our own instance of the pre aggregator
-			@SuppressWarnings("unchecked")
-			ReduceFunction<T> functionCopy = (ReduceFunction<T>) SerializationUtils.clone(function);
```

Comment:
The user function will be serialized for shipping to the TaskManagers. 
So at the Taskmanager's, will create new instances of the user function